src/mono/wasm/runtime/exports.ts

```diff
@@ -293,6 +294,10 @@ function initializeImportsAndExports(
 
     configure_emscripten_startup(module, exportedAPI);
 
+    // HACK: Maintain compatibility with emscripten's generated dotnet.worker.js
```

Comment:
this is my fault, but we should probably expand this comment before merging it. maybe:

```
// HACK: Emscripten expects the return value of this function to always be the Module object, but
//  we changed ours to return a set of exported namespaces. In order for the emscripten generated
//  worker code to keep working, we detect that we're running in a worker (via the presence of 
//  globalThis.importScripts) and emulate the old behavior.
// Note that this will impact anyone trying to load us in a web worker directly, not just emscripten!
```